pkg/inject/inject.go

```diff
@@ -491,6 +493,11 @@ func (conf *ResourceConfig) complete(template *corev1.PodTemplateSpec) {
 
 // injectPodSpec adds linkerd sidecars to the provided PodSpec.
 func (conf *ResourceConfig) injectPodSpec(values *patch) {
+	opaquePorts, err := conf.proxyOpaquePorts()
+	if err != nil {
+		log.Warnf("unrecognized value used for the %s annotation: %s", k8s.ProxyOpaquePortsAnnotation, conf.getOverride(k8s.ProxyOpaquePortsAnnotation))
```

Comment:
NIT: maybe it is worth logging the error as well, as it might provide some more insight into what went wrong with parsing for example